### PR TITLE
add OpenSCAD language support

### PR DESCRIPTION
**Pull request by kenchou**
OpenSCAD is a software for creating solid 3D CAD objects.
SCAD stands for "Scripted CAD", where CAD is an acronym for "Computer-Aided Design".

-  https://github.com/openscad/openscad
-  https://openscad.org/documentation.html#language-reference

OpenSCAD has been added to the upstream repository Tokei ([v13.0.0-alpha.2](https://github.com/XAMPPRocky/tokei/releases/tag/v13.0.0-alpha.2))  ([ PR #1097](https://github.com/XAMPPRocky/tokei/pull/1097))

Due to Perl 6 being renamed to Raku in Tokei v13.x, this PR has also been updated accordingly to avoid build errors.

### CODE REVIEW

File: languages.yaml [hunk 1]
```diff
@@ -1788,6 +1788,39 @@ Odin:
...
+      - magenta
+      - magenta
+      - white
+    hex: # optional, alternative to basic colors for terminals that support true colour.
+      - "#9B4F97"
+      - "#67217A"
+      - "#803788"
+      - "#FFFFFF"
+    chip: "#178600" # required, this is used for the language breakdown bar, its value can be found in linguist (link 2).
+    icon: '\u{F34E}'
```
Comment (o2sh) [inline languages.yaml:1823]: The `icon` property's indentation needs to be decreased by one level.

This is why the Nerd Font icon isn't appearing in the web preview. https://vercel-cxhqbujdb-o2shs-projects.vercel.app/#OpenScad

File: languages.yaml [hunk 2]
```diff
@@ -1788,6 +1788,39 @@ Odin:
...
+    ansi: # required, a list of the ANSI colors used to colorize the logo
+      - blue
+      - magenta
+      - magenta
+      - white
+    hex: # optional, alternative to basic colors for terminals that support true colour.
+      - "#9B4F97"
+      - "#67217A"
+      - "#803788"
+      - "#FFFFFF"
```
Comment (o2sh) [inline languages.yaml:1821]: You're declaring 4 colors but only using one.

More info on how to color an ascii logo: https://github.com/o2sh/onefetch/wiki/ascii-art

File: languages.yaml [hunk 3]
```diff
@@ -1788,6 +1788,39 @@ Odin:
...
+    {0}-.    .:.    -************=        *.
+    {0}=       .:.   :**********-        :+
+    {0}=         :.   =********=        .+.
+    {0}.-         -   .********-       :=.
+    {0} .-.       :   =********+.   .:=-
+    {0}   ::     .-:=************++==:
+    {0}     ·...·   .-=++*****++=-.
+    {0}                ---------
+  colors:
+    ansi: # required, a list of the ANSI colors used to colorize the logo
```
Comment (o2sh) [inline languages.yaml:1812]: you can remove the comments

File: languages.yaml [hunk 4]
```diff
@@ -1887,7 +1920,7 @@ Perl:
       - cyan
     chip: "#0298C3"
   icon: '\u{E67E}'
-Perl6:
+Raku:
```
Comment (spenserblack) [inline languages.yaml:1923]: The languages are alphabetized, so this would need to be moved.

File: Cargo.toml [hunk 5]
```diff
@@ -66,7 +66,7 @@ serde_yaml = "0.9.34"
 strum.workspace = true
 time = { version = "0.3.37", features = ["formatting"] }
 time-humanize = { version = "0.1.3", features = ["time"] }
-tokei = "12.1.2"
+tokei = "13.0.0-alpha.7"
```
Comment (spenserblack) [inline Cargo.toml:69]: @o2sh So we're good with moving to a pre-release? I don't have a problem with it, but what are your thoughts?

**Review by spenserblack — CHANGES_REQUESTED** [review]

File: Cargo.toml (hunk 5 above)
Comment (o2sh) [reply Cargo.toml:69]: It's definitely not ideal, but we can always roll back if we find any issues

File: languages.yaml (hunk 1 above)
Comment (kenchou) [reply languages.yaml]: fixed

File: languages.yaml (hunk 2 above)
Comment (kenchou) [reply languages.yaml]: fixed. Removed unnecessary color definitions.

File: languages.yaml (hunk 3 above)
Comment (kenchou) [reply languages.yaml]: fixed

File: languages.yaml (hunk 4 above)
Comment (kenchou) [reply languages.yaml]: fixed. Moved to after `Racket`

File: Cargo.toml (hunk 5 above)
Comment (kenchou) [reply Cargo.toml:69]: Sadly, the upstream release is too slow. It took 3 months from merging this PR to the release. I almost forgot about it.

File: languages.yaml (hunk 2 above)
Comment (o2sh) [reply languages.yaml]: [94b5656](https://github.com/o2sh/onefetch/pull/1502/commits/94b565681ce3d352880d7b4b56ee1cf857bb5cf2) I added a second color with hex values to get it closer to the original logo https://vercel-f4f8n13fs-o2shs-projects.vercel.app/#OpenScad

**Review by o2sh — APPROVED** [review]

**Review by spenserblack — APPROVED** [review]
Because I'm picky about git histories, ideally this PR would be 1 commit to bump tokei and 1 to add OpenSCAD support, but I'm not going to block based on that. LGTM!

**Comment by o2sh** [comment]
Good point @spenserblack, I bumped tokei in its own commit da2363518b38c351e1e5824563451d42d3d715f3 and reverted the corresponding changes from this PR

**Review by o2sh — APPROVED** [review]
Thanks a lot for your contribution @kenchou, the ASCII logo looks great 🎉

**Pull request merged**